### PR TITLE
Fix for maybe_placeholders when keep_all_tokens=True

### LINKED ISSUE #706: maybe_placeholders and anonymous terminal omits the None
Doc says [here](https://lark-parser.readthedocs.io/en/latest/grammar.html#rules) that `maybe_placeholders=True` makes ` rule : some1 [item] some2` to produce a `None` if item is not found, but anonymous token seems to not produce `None`.

Minimal reproduction: 
~~~python
from lark import Lark

grammar = r"""
UINT : /\d+/

math_atom : UINT 

LESS : "<"

math_script : math_atom ("_" ["<"] math_atom)*
"""

parser = Lark(
    grammar= grammar,
    parser="lalr",
    keep_all_tokens=True,
    maybe_placeholders=True,
    start="math_script"
     )

test_file= "1_1"
tree = parser.parse(test_file)
print(tree.pretty())
~~~

Produces: 
~~~
math_script
  math_atom     1
  _
  math_atom     1
~~~

If you replace `math_atom` to `UINT` it works. 
if you replace `["<"]` with `[LESS]` it works. 

~~~
math_script
  math_atom     1
  _
  None
  math_atom     1
~~~

I have both a global lark installed and a fresh environment with just lark

global install `Version: 0.9.0`
environment install  `Version: 0.10.0`

### PR DESCRIPTION
**Pull request by MegaIng**
Fixes #706.

Includes test

slight change in the design: now the GrammarLoader is reused when importing, which shouldn't make a difference.

### CODE REVIEW

File: lark/load_grammar.py [hunk 1]
```diff
@@ -946,7 +947,10 @@ def load_grammar(self, grammar_text, grammar_name='<?>'):
         rules = rule_defs
 
         rule_names = {}
-        for name, params, _x, _o in rules:
+        for name, params, _x, option in rules:
+            if self.always_keep_all_tokens: # We need to do this somewhere. Might as well prevent an additional loop
```
Comment (erezsh) [inline lark/load_grammar.py:951]: This comment isn't helpful.

**Comment by erezsh** [comment]
Everything looks alright. I'd just remove that comment, or replace it with something that new readers might find useful.


Does it conflict with #710? I don't see how.

**Comment by MegaIng** [comment]
It changes the placement of the usage of `keep_all_tokens`, meaning the `_load` does not have a chance to correct the rules I think. Previously it was only used in building the callbacks which is done in `_load`, but now it is done in `load_grammar`. I think the best solution for this is extending #710/creating a new PR that allows options that need to be specified early like `keep_all_tokens` (after this PR (#707)) or `maybe_placeholders` to be set from the command line.

**Comment by erezsh** [comment]
Why would `_load` need to correct the rules? Aren't the `RuleOptions` saved and loaded together with the rules?

**Comment by MegaIng** [comment]
Yes, but we have a difference between the options before and after loading. That is the core problem here. When the standalone parser is generated, `keep_all_tokens` is not set, therefore `load_grammar` does not set the attributes correctly and the wrong attributes are saved. When afterwards `keep_all_tokens` is set, this will not be reflected in the attributes of the `RuleOptions`. (And just modifying the RuleOptions afterwards once again doesn't help for `maybe_placeholders` and is just a hack)

**Comment by erezsh** [comment]
I see. So that leaves us with two options:

1. Require `keep_all_tokens` to be set when creating the standalone, and forbid changing it afterwards

2. Revert to the original code where `keep_all_tokens` only affected the tree builder

Unless I missed something?

**Comment by MegaIng** [comment]
> Revert to the original code where keep_all_tokens only affected the tree builder

Yes, but then we also need to redesign `maybe_placeholders`. While possible, this is probably something you should take on as I don't know how it really works.

**Comment by erezsh** [comment]
Yes, I think 1 is the best option. It's also not such a popular switch in the first place.

**Comment by MegaIng** [comment]
The cosmic rays decided that the tests shall not run.

**Comment by erezsh** [comment]
The universe must have its reasons.

**Pull request merged**